### DISCUSSION TITLE
Full food101 dataset accuracy at 75% with effnetb2

### DISCUSSION THREAD

**Discussion by longtongster** [mrdbourke/pytorch-deep-learning#460] [General]
Was anyone able to get a higher accuracy on the full food101 dataset test set? If so can you please share your ideas? I am looking to learn tips and tricks to further improve, if possible.

I just did some straightforward training of the head of 7 epochs and then some finetuning also for 7 epochs on 30% of the training data. Looking at my loss and accuracy the test acc is still higher than train accuracy so the model has some capacity to further improvement. I see that in finetuning certain layers need to be kept frozen (batchnorm).  Maybe a suggestions for @mrdbourke to include a bit more details on how exactly to do fine-tuning in the course. Going to the full dataset we have too much data just to only train the classifier so it makes sense to move towards fine-tuning. 
   
Kind Regards
Longtongster

**Comment by mrdbourke** [comment]
Hi @longtongster ,

Good to see you're trying to improve your models!

As for fine-tuning models, I'd say your next step is to try fine-tuning all the layers with code such as: 

```python
for param in model.parameters():
            param.requires_grad = True
```

This means all the parameters in the target `model` will have their gradients tracked and updated in the training code.

A good tip while fine-tuning is to change the learning rate to 10x smaller (because the weights are already trained on an existing dataset, you don't want to overtrain them), for example go from `lr=0.001` -> `lr=0.0001`.

I'd try doing this for ~3-5 epochs and see how it goes.

**Reply by longtongster** [reply]
Thanks for sharing the idea @mrdbourke . I was doing already some fine-tuning but found that with a learning rate scheduler and a bit more training I am able to get to increase to 81% accuracy with the efficient net b2 with in the order of 8 mio parameters. Have you gone for the Full Monty and tried to squeeze more out of this dataset? If you have more ideas please let me know.

I also looked at some of the most confident wrong predictions and feel like I would have not been able to guess the real label. Some are obviously wrong. The model has quite some cheese cakes wrong once there is also a strawberry in the image :-)

**Comment by longtongster** [comment]
Just an update for anyone interested ... with using 40% of the data, I am able to get to an accuracy of 84.9% on the full test set by applying just fine-tuning for 10 epochs and a linearLR learning rate schedular that starts at 1e-4. On my machine that takes quite some time (e.g 1 hour). Some images have a meal with specific ingredients then the model might predict the for example the ingredients with highest prob and not the meal (which is the label).

**Reply by mrdbourke** [reply]
Nice work @longtongster !

I find that too, if a meal has more than one possible label (e.g. the image contains a burger and fries) the model often predicts both classes in the top 1-3 labels. 

That's one of the tradeoffs of single-class classification, models can get very good but often images contain more than one item.
